### ISSUE TITLE
R1732(consider-using-with) false positive

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#5080]
### Bug description

I use `open("foo","w").close()` as a [`touch`](https://en.wikipedia.org/wiki/Touch_(command)) idiom to create an empty file.
I did not see any diagnostic about this before.

I think that `consider-using-with` note is not appropriate for such a line because the resource is being released as it is allocated.

code:
```python
"https://github.com/PyCQA/pylint/issues/5080"
open("foo","w").close()
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:2: [R1732(consider-using-with), ] Consider using 'with' for resource-allocating operations
a.py:2: [W1514(unspecified-encoding), ] Using open without explicitly specifying an encoding
```

### Expected behavior

no diagnostics

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.9.7 (default, Sep  3 2021, 12:45:31) 
[Clang 12.0.0 (clang-1200.0.32.29)]
```

**Comment by sam-s** [comment]
there seems to be a work-around:
```
with open("foo"): pass
```
but the message is still wrong.

**Comment by DudeNr33** [comment]
I agree that this example of opening and closing in the same line is something where the main concerns of the check do not really apply.
However, this would not be easy to fix and maintain, as the ``consider-using-with`` check also targets other functions and methods besides just ``open()``. 
Thus, ``pylint`` would need to track all the corresponding functions / methods that release the resource. 

Also, as you mentioned, there a "workarounds" (which I would consider perfectly fine code). If you want something more explicit, you can even use the ``pathlib.Path.touch()`` method from the standard library.

If you want to stick to your current implementation, you can also disable this check for this particular line, or altogether via config file or command line.

**Comment by Pierre-Sassoulas** [comment]
Closing as won't do because it's both high effort and minor.
